atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -241,7 +252,44 @@ public String toString() {
         poolConfig.setTestWhileIdle(true);
 
         poolConfig.setJmxNamePrefix(host.getHostString());
+        GenericObjectPool<Client> pool = new GenericObjectPool<>(cassandraClientFactory, poolConfig);
+        registerMetrics(pool, host.getHostString());
+        return pool;
+    }
+
+    private void registerMetrics(GenericObjectPool<Client> pool, String metricPrefix) {
+        registerMetric(pool, metricPrefix, "-mean-active-time-millis", pool::getMeanActiveTimeMillis);
+        registerMetric(pool, metricPrefix, "-mean-idle-time-millis", pool::getMeanIdleTimeMillis);
+        registerMetric(pool, metricPrefix, "-mean-borrow-wait-time-millis", pool::getMeanBorrowWaitTimeMillis);
+        registerMetric(pool, metricPrefix, "-proportion-destroyed-by-evictor",
+                () -> {
+                    long created = pool.getCreatedCount();
+                    long evicted = pool.getDestroyedByEvictorCount();
+                    return (double) evicted / (double) created;
+                });
+        registerMetric(pool, metricPrefix, "-proportion-destroyed-by-borrower",
+                () -> {
+                    long created = pool.getCreatedCount();
+                    long evicted = pool.getDestroyedByBorrowValidationCount();
+                    return (double) evicted / (double) created;
+                });
+    }
 
-        return new GenericObjectPool<>(cassandraClientFactory, poolConfig);
+    private void registerMetric(GenericObjectPool<Client> pool, String metricPrefix, String metricName, Gauge gauge) {
+        if (!metricRegistry.isPresent()) {
+            return;
+        }
+        String fullyQualifiedMetricName = MetricRegistry.name(
+                CassandraClientPoolingContainer.class, metricPrefix + metricName);
```

Comment:
Can avoid the string concatenation here and fall back on varargs method, though should make sure we don't append dot on prefix